premium/ecommerce/gdpr-send-to-mailchimp.php

```diff
@@ -5,38 +5,99 @@
  */
 
 /**
- * Send a request to mailchimp to check if member has status subscribed or pending
+ * Check if a list member has status pending or subscribed based on there email
  *
  * @see https://mailchimp.com/developer/reference/lists/list-members/
  *
  * @param string $email
  *
  * @return bool
  */
-function sync_only_subscribed_or_pending_emails($email) {
+function mailchimp_list_member_has_status_pending_or_subscribed($email) {
+
 	if(!is_string($email)) {
 		throw new InvalidArgumentException(
-			sprintf("sync_only_subscribed_or_pending_emails() expects parameter 1 to be array or %s, %s given", Traversable::class, gettype($emails))
+			sprintf("mailchimp_list_member_has_status_pending_or_subscribed() expects parameter 1 to be string, %s given", gettype($email))
 		);
 	}
 
-	if(!mc4wp()->has('ecommerce.options')) {
-		return false;
-	}
+    if(!mc4wp()->has('ecommerce.options')) {
```

Comment:
Since this function is only ever called from within the filter callbacks which in turn are never fired from anywhere but the e-commerce3 plugin, this check is redundant. If the service does not exist this will throw an exception, which is probably what we want if users are calling this themselves without having e-commerce set-up. So I think we should remove these 3 lines.